scipy/sparse/sparsetools/coo.h

```diff
@@ -112,6 +112,43 @@ void coo_todense(const I n_row,
 }
 
 
+template <class I, class T>
+void coo_todense_nd(const I D[],        // array of dimension sizes
```

Comment:
The term "dimension sizes" is called "shape" now (though maybe some older c++ code used that term).
I also prefer the variable name `shape` over `D` for readability.  I'm also not sure what the name `A` refers to in this function. The code uses it as the coords array, so maybe name it something to indicate that.

Similar name suggestions for other functions (and these are just suggestions. If you think D is more readable just say so).

Reply:
Changing `D` to `shape` sounds like a reasonable choice! As for `A`, I agree that the name isn't indicating anything. Since it is the coords array, for uniformity with `Ax` for data array (which I took from `coo_todense`), I could rename `A` to `Ac`. If that's not as sensible either, then we could change `Ac` to `A_coords` and `Ax` to `A_data` (or something similar). The commit at 5823bbe used `Ac`, but if you recommend `A_coords`, then I can make the changes!

Reply:
Some of the other functions use `Ai` and `Aj` names for the coords in 2D. Perhaps this could be `Aijk`.
I don't think `Ac` is better than `A`. But `Acoo` or `A_coords` would be fine.  The `Ax` is used in **lots** of functions in `sparsetools`, so I think you should keep that as is.

Reply:
Changed to `Aijk` at 8f4075f.